build-logic/src/main/kotlin/org.sonarsource.php.java-conventions.gradle.kts

```diff
@@ -0,0 +1,36 @@
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat
+import org.gradle.api.tasks.testing.logging.TestLogEvent.FAILED
+import org.gradle.api.tasks.testing.logging.TestLogEvent.SKIPPED
+
+plugins {
+  `java-library`
+  jacoco
+}
+
+java {
+  withSourcesJar()
+  withJavadocJar()
+  sourceCompatibility = JavaVersion.VERSION_17
+  targetCompatibility = JavaVersion.VERSION_17
+}
+
+tasks.withType<JavaCompile> {
+  options.encoding = "UTF-8"
+}
+
+tasks.withType<Javadoc> {
+  options.encoding = "UTF-8"
+  options {
+    (this as CoreJavadocOptions).addStringOption("Xdoclint:none", "-quiet")
+  }
+}
+
+tasks.withType<Test> {
+  useJUnitPlatform()
+  testLogging {
+    // log the full stack trace (default is the 1st line of the stack trace)
+    exceptionFormat = TestExceptionFormat.FULL
+    // verbose log for failed and skipped tests (by default the name of the tests are not logged)
+    events(SKIPPED, FAILED)
+  }
+}
```

Comment:
There is "no data about coverage" because the PR doesn't contain any new Java code. Looking at this test PR the code coverage seems to work properly: https://github.com/SonarSource/sonar-php/pull/1295

Reply:
"no data about coverage" - It is somehow misleading

Reply:
Indeed 🤔 I guess it gets nothing from JaCoCo so it does not differentiate between missing data and no data required